### PR TITLE
Hw 22

### PR DESCRIPTION
**Pull request by reemaRahhal**
Hi Elias, 
I finished my task regarding question password validition,
I used in ascii charecter code to check if the characters that user entered are in the range accordingly. 
Regarding the question of sorting numbers's array , in academic studies, I studied several types of sorting and 
used one of them is called counting sorting which time's complecation is  O(n). 
can you review my task please .

### CODE REVIEW

File: Home_Work_22/js/sort.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+  }
+
+  // create a new array to store a sorting numbers
+
+  var newArr = [];
+  for (i in counter) {   //------------------------O(n)--------------------
+    while (counter[i] > 0) {
+      newArr.push(parseInt(i));
+      counter[i]--;
```
Comment (EliasSalom) [inline Home_Work_22/js/sort.js:27]: in this loop, there is a `while` inside `for` so this will make the function be `O(n^2)`

**Review by EliasSalom — CHANGES_REQUESTED** [review]
please check my comment and change it

File: Home_Work_22/js/sort.js (hunk 1 above)
Comment (reemaRahhal) [reply Home_Work_22/js/sort.js]: Hi Elias, 
I changed the sorting  array  function, please review my task

**Review by EliasSalom — APPROVED** [review]
have you solved this task alone?

**Pull request merged**